lib/ansible/modules/identity/cyberark/cyberark_authentication.py

```diff
@@ -0,0 +1,300 @@
+                    'supported_by': 'community',
+                    'metadata_version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: cyberark_authentication
+short_description: "Module for CyberArk Vault Authentication using Privileged Account Security Web Services SDK"
+author: "Edward Nunez (@enunez-cyberark)"
+version_added: "2.4"
+description:
+    - "Authenticates to CyberArk Vault using Privileged Account Security Web Services SDK and
+       creates a session fact that can be used by other modules. It returns an Ansible fact
+       called I(cyberark_session). Every module can use this fact as C(cyberark_session) parameter."
+
+
+options:
+    state:
+        default: present
+        choices: ['present', 'absent']
+        description:
+            - Specifies if an authentication logon/logoff and a cyberark_session should be added/removed.
+    username:
+        description:
+            - The name of the user who will logon to the Vault.
+    password:
+        description:
+            - The password of the user.
+    new_password:
+        description:
+            - The new password of the user. This parameter is optional, and enables you to change a password.
+    api_base_url:
+        description:
+            - A string containing the base URL of the server hosting CyberArk's Privileged Account Security Web Services SDK.
+    validate_certs:
+        default: true
+        description:
+            - If C(false), SSL certificates will not be validated.  This should only
+              set to C(false) used on personally controlled sites using self-signed
+              certificates.
+    use_shared_logon_authentication:
+        default: false
+        description:
+            - Whether or not Shared Logon Authentication will be used.
+    use_radius_authentication:
+        default: false
+        description:
+            - Whether or not users will be authenticated via a RADIUS server. Valid values are true/false.
+    cyberark_session:
+        description:
+            - Dictionary set by a CyberArk authentication containing the different values to perform actions on a logged-on CyberArk session.
+'''
+
+EXAMPLES = '''
+- name: Logon to CyberArk Vault using PAS Web Services SDK - use_shared_logon_authentication
+  cyberark_authentication:
+    api_base_url: "{{ web_services_base_url }}"
+    use_shared_logon_authentication: true
+
+- name: Logon to CyberArk Vault using PAS Web Services SDK - Not use_shared_logon_authentication
+  cyberark_authentication:
+    api_base_url: "{{ web_services_base_url }}"
+    username: "{{ password_object.password }}"
+    password: "{{ password_object.passprops.username }}"
+    use_shared_logon_authentication: false
+
+- name: Logoff from CyberArk Vault
+  cyberark_authentication:
+    state: absent
+    cyberark_session: "{{ cyberark_session }}"
+'''
+
+RETURN = '''
+cyberark_session:
+    description: Authentication facts.
+    returned: success
+    type: dict
+    sample:
+        {
```

Comment:
@bcoca I changed the documentation to describe the individual subkeys of the cyberark_session key, instead of the example data I had before.